docs/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,227 @@
+# Rally Release Process
```

Comment:
@badboy any chance you could sanity-check this process to see if it makes sense and if the constraints are clear from an outsider POV?